### ISSUE TITLE
Add pre-signed URL request logging to server

### ISSUE THREAD

**Issue by btello** [delta-io/delta-sharing#39]
Currently there's not a great way to understand how many pre-signed URLs are being generated by the sharing server.  It would be nice if the server logged at the DEBUG level every time it generated a new pre-signed S3 URL.  Having access to these logs would aid in performance testing and debugging.

**Comment by zsxwing** [comment]
Yep. These logs would be helpful. It's probably too verbose if we output one log for each URL. But we can output some aggregated logs, such as, `Took XXX ms to sign XXX urls for table YYY`
